raiden/blockchain/state.py

```diff
@@ -21,9 +20,14 @@ def get_channel_state(
         reveal_timeout: BlockTimeout,
         payment_channel_proxy,
         opened_block_number: BlockNumber,
-        opened_block_hash: BlockHash,
 ):
-    channel_details = payment_channel_proxy.detail(opened_block_hash)
+    # Here we have to query the latest state as during restart a lot of things
+    # can have happened, such as the channel having been closed. To get this
+    # information we should use the latest block hash
```

Comment:
The amount of stuff that happened in between should not matter. This is used to get the channel state to be used in a channel open event, eventually the channel close event will also be processed.

It would just be easier to understand if you mention that this is avoiding querying state from an old block because that may fail.

Reply:
The reason the comment mentions channel open is because this is the only block identifier passed to this function and this is what was being used before and we need to explain why this should not be used anymore.

Reply:
Hummm, I think we are talking about different things.

What I mean is that this function is *only used for the ChannelOpened event processing*:

```sh
# git rev-parse HEAD
09e422b610767c0c9a3d85da00f48038ed3406c8
# ag --python get_channel_state                                                                                                                     12:58
raiden/network/proxies/token_network.py
327:            channel_state = self._get_channel_state(
500:            channel_state = self._get_channel_state(
519:            channel_state = self._get_channel_state(
538:            channel_state = self._get_channel_state(
858:            channel_state = self._get_channel_state(
1481:    def _get_channel_state(
1512:        channel_state = self._get_channel_state(

raiden/blockchain_events_handler.py
7:from raiden.blockchain.state import get_channel_state
93:        channel_state = get_channel_state(

raiden/blockchain/state.py
17:def get_channel_state(
```

And if the state of the channel is anything else doesn't matter, because the events like `ChannelClosed` will be processed afterwards.